Common/src/geometry/CGeometry.cpp

```diff
@@ -2456,16 +2456,73 @@ su2double CGeometry::GetSurfaceArea(const CConfig* config, unsigned short val_ma
    * All nodes that are shared by multiple symmetries have to get a corrected normal. */
+
+  symmetryNormals.clear();
   symmetryNormals.resize(nMarker);
   std::vector<unsigned short> symMarkers;
 
+  /*--- Compute if marker is a straight line or plane. ---*/
+  ComputeSurf_Straightness(config, false);
+
+  /*--- Check which markers are symmetry or Euler wall and store in a list. ---*/
   for (auto iMarker = 0u; iMarker < nMarker; ++iMarker) {
     if ((config->GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE) ||
         (config->GetMarker_All_KindBC(iMarker) == EULER_WALL)) {
       symMarkers.push_back(iMarker);
     }
   }
 
+  /*--- Loop over all markers and find nodes shared on curved symmetry/Euler markers. ---*/
+  for (size_t i = 1; i < symMarkers.size(); ++i) {
+    const auto iMarker = symMarkers[i];
+    for (auto iVertex = 0ul; iVertex < nVertex[iMarker]; iVertex++) {
+      const auto iPoint = vertex[iMarker][iVertex]->GetNode();
+
+      /*--- Halo points do not need to be considered. ---*/
+      if (!nodes->GetDomain(iPoint)) continue;
+
+      /*--- Get the vertex normal on the current symmetry. ---*/
+      std::array<su2double, MAXNDIM> iNormal = {};
+      std::array<su2double, MAXNDIM> kNormal = {};
+      vertex[iMarker][iVertex]->GetNormal(iNormal.data());
+      const su2double iarea = GeometryToolbox::Norm(nDim, iNormal.data());
+      for (auto iDim = 0u; iDim < nDim; iDim++) iNormal[iDim] /= iarea;
+
+      /*--- Loop over previous symmetries and if this point shares them, sum the normals. ---*/
+      for (size_t j = 0; j < i; ++j) {
+        const auto jMarker = symMarkers[j];
+        const auto jVertex = nodes->GetVertex(iPoint, jMarker);
+        if (jVertex < 0) continue;
+
+        /*--- If both symmetries are curved, we sum the normals, else continue the search. ---*/
+        if ((bound_is_straight[iMarker] == true) || (bound_is_straight[jMarker] == true)) continue;
+
+        std::array<su2double, MAXNDIM> jNormal = {};
+        const auto it = symmetryNormals[jMarker].find(jVertex);
+
+        if (it != symmetryNormals[jMarker].end()) {
+          jNormal = it->second;
+        } else {
+          vertex[jMarker][jVertex]->GetNormal(jNormal.data());
+          const su2double jarea = GeometryToolbox::Norm(nDim, jNormal.data());
+          for (auto iDim = 0u; iDim < nDim; iDim++) jNormal[iDim] /= jarea;
+        }
+
+        /*--- Average the normals on the shared nodes. Also normalize them again. ---*/
+        for (auto iDim = 0u; iDim < nDim; iDim++) kNormal[iDim] = iNormal[iDim] + jNormal[iDim];
+        const su2double karea = GeometryToolbox::Norm(nDim, kNormal.data());
+        for (auto iDim = 0u; iDim < nDim; iDim++) kNormal[iDim] /= karea;
+
+        for (auto iDim = 0u; iDim < nDim; iDim++) {
+          symmetryNormals[iMarker][iVertex][iDim] += kNormal[iDim];
+          symmetryNormals[jMarker][jVertex][iDim] += kNormal[iDim];
+        }
```

Comment:
+=? not just =? What if there are more than 2 markers?

Reply:
ok, I will now first add all normals (n_i + n_j) for all markers here, then += is correct. I then normalize after this routine.